packages/dashboard/src/components/tasks/utils.ts

```diff
@@ -2,46 +2,66 @@ import { SubmitTask } from 'api-client';
 import * as RmfModels from 'rmf-models';
 
 /* istanbul ignore next */
-function checkField(obj: Record<string, unknown>, field: string, type: string): void {
-  if (!Object.prototype.hasOwnProperty.call(obj, field) || typeof obj[field] !== type) {
-    throw new TypeError(`expected [${field}] to be [${type}]`);
+function checkField(
```

Comment:
Suggest to remove `errMsgs` param and wrap this in a `checkFile` function. The function will keep its own list of error messages and call `checkFields` with try/catch.

On another node, the try/catch here is useless because you are throwing and immediately catching any exceptions, you can just call `errMsgs.push` without any try catch.